assets/js/components/ExecutionResults/ExecutionResults.jsx

```diff
@@ -15,65 +13,79 @@ import {
   ResultsContainer,
   HostResultsWrapper,
   CheckResult,
-  getHosts,
   getChecks,
   getHealth,
   getCheckResults,
   getCheckDescription,
 } from '@components/ChecksResults';
 import { UNKNOWN_PROVIDER } from '@components/ClusterDetails/ClusterSettings';
+import NotificationBox from '@components/NotificationBox';
 
 const truncatedClusterNameClasses =
   'font-bold truncate w-60 inline-block align-top';
 
-const getLabel = (health, expectations, failedExpectations) =>
-  health === 'passing'
-    ? `${expectations}/${expectations} expectations passed`
-    : `${failedExpectations}/${expectations} failed`;
+const getLabel = (status, health, error, expectations, failedExpectations) => {
+  if (status === 'running') {
+    return '';
+  }
+
+  if (error) {
+    return error;
+  }
+
+  if (health === 'passing') {
+    return `${expectations}/${expectations} expectations passed`;
+  }
+
+  return `${failedExpectations}/${expectations} expectations failed`;
+};
 
 function ExecutionResults({
   clusterID,
-  executionID,
   clusterName,
   cloudProvider,
   hostnames = [],
-  onExecutionFetch = getExecutionResult,
-  onCatalogFetch = getCatalog,
+  catalogLoading,
+  catalog,
+  catalogError,
+  executionLoading,
+  executionData,
+  executionError,
   onCatalogRefresh = () => {},
+  onLastExecutionUpdate = () => {},
 }) {
-  const [loading, setLoading] = useState(false);
-  const [executionData, setExecutionData] = useState(null);
-  const [catalog, setCatalog] = useState(null);
   const [selectedCheck, setSelectedCheck] = useState(null);
   const [modalOpen, setModalOpen] = useState(false);
 
   useEffect(() => {
-    setLoading(true);
-    Promise.all([onExecutionFetch(executionID), onCatalogFetch()])
-      .then(
-        ([{ data: fetchedExecutionData }, { data: fetchedCatalogData }]) => {
-          setLoading(false);
-          setExecutionData(fetchedExecutionData);
-          setCatalog(fetchedCatalogData.items);
-        }
-      )
-      .catch((error) => {
-        setLoading(false);
-        logError(error);
-      });
-  }, [onExecutionFetch, onCatalogFetch, setExecutionData, setCatalog]);
-
-  if (loading) {
+    onCatalogRefresh();
+    onLastExecutionUpdate();
+  }, []);
+
+  if (catalogLoading || executionLoading) {
     return <LoadingBox text="Loading checks execution..." />;
   }
 
+  if (catalogError || executionError) {
+    return (
+      <NotificationBox
+        icon={<EOS_ERROR className="m-auto" color="red" size="xl" />}
+        text={
+          catalogError && executionError
+            ? `${catalogError}\n${executionError}`
+            : catalogError || executionError
+        }
+        buttonText="Try again"
+        buttonOnClick={onCatalogRefresh}
+      />
+    );
+  }
+
   if (executionData?.status === 'running') {
```

Comment:
We currently render a `LoadingBox` spinner if `status === 'running'`, isn't that the intended behaviour?

Reply:
No. The intention is to show all the individual checks with the spinner icon

Reply:
@jamie-suse in order to do so we can resuse the `getChecks` function as it was. It's a trick for me because with the current logic preserving that behavior has little value, but I don't have hard feelings about that

Reply:
And removing simply this `if` statement doesn't do the trick?

Reply:
I honestly should try :smile:

Reply:
@arbulu89 on a second thought: I don't remember if a response from Wanda with a `running` execution includes any check at all cause the field is nullable

https://github.com/trento-project/wanda/blob/main/lib/wanda_web/schemas/execution_response.ex

Reply:
It doesn't include any check results. But we have the requested checks for each target in the targets field. So we can populate them easily. I think that removing that `if` condition is enough.
We already made it work when we paired on that with @EMaksy @rtorrero , etc

Reply:
Ah yes i think Xabi is fully right. 
We did it in the other branch with @rtorrero
This kinda got lost in the process! 
See here: https://github.com/rtorrero/web/commit/d892254ca9d38365b70938c26bff8e3cc4659108#diff-a5da6e4a9398f27bbce1e47fba23dfeb0a8b44cec0a5b377f6d604cb41090d03R70

Reply:
Fair enough :+1: feel free to restore this then :smile: